cloudstate-kotlin-support/src/main/kotlin/io/cloudstate/kotlinsupport/api/eventsourced/EventSourcedHandler.kt

```diff
@@ -0,0 +1,204 @@
+package io.cloudstate.kotlinsupport.api.eventsourced
+
+import com.google.protobuf.GeneratedMessageV3
+import io.cloudstate.javasupport.Context
+import io.cloudstate.kotlinsupport.logger
+import org.apache.commons.lang3.builder.ToStringBuilder
+import kotlin.reflect.KParameter
+import kotlin.reflect.KType
+import kotlin.reflect.jvm.reflect
+
+open class EventSourcedHandler<S : Any> {
+    private val log = logger()
+
+    private val methodDefinitions = mutableListOf<MethodDefinition>()
+
+    var state: S? = null
+    var persistenceId: String? = null
+    var snapshotEvery: Int? = 0
+
+    fun getMethodDefinitions() = methodDefinitions
+
+    inline fun withState(initialState: S, block: EventSourcedHandler<S>.() -> EventSourcedHandler<S>) = apply {
```

Comment:
Please, don't use an `inline` method on public API.

Reply:
Thanks for the comments @ihostage, sorry, but I only see it now.
***withState*** will be completely removed, I send the PR just so Viktor can test some problems. This API was just a local draft and will be removed.